code/datums/quirks/positive_quirks.dm

```diff
@@ -362,3 +362,165 @@
 
 /datum/quirk/item_quirk/signer/remove()
 	qdel(quirk_holder.GetComponent(/datum/component/sign_language))
+
+/datum/quirk/spacer_born
+	name = "Spacer"
+	desc = "You were born in space, and have never known the comfort of a planet's gravity. Your body has adapted to this. \
+		You are more comfortable in zero and artifical gravity and are more resistant to the effects of space, \
+		but travelling to a planet's surface for an extended period of time will make you feel sick."
+	icon = FA_ICON_USER_ASTRONAUT
+	value = 7
+	quirk_flags = QUIRK_HUMAN_ONLY|QUIRK_CHANGES_APPEARANCE
+
+	/// How long on a planet before we get averse effects
+	var/planet_period = 3 MINUTES
+	/// TimerID for time spend on a planet
+	var/planetside_timer
+	/// How long in space before we get beneficial effects
+	var/recover_period = 1 MINUTES
+	/// TimerID for time spend in space
+	var/recovering_timer
+
+/datum/quirk/spacer_born/add(client/client_source)
+	var/mob/living/carbon/human/human_quirker = quirk_holder
+	human_quirker.set_mob_height(HUMAN_HEIGHT_TALLER)
+	human_quirker.physiology.pressure_mod *= 0.8
+
+	// Only start tracking when we arrive on station (or a mining level of a map is weird)
+	var/spawn_z = quirk_holder.loc?.z
+	if(is_station_level(spawn_z) || is_mining_level(spawn_z))
+		start_tracking()
+	else
+		RegisterSignal(quirk_holder, COMSIG_MOVABLE_MOVED, PROC_REF(arrived_at_station))
+
+/datum/quirk/spacer_born/remove()
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED)
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_MOVED)
+
+	if(!QDELING(quirk_holder))
+		quirk_holder.clear_mood_event("spacer")
+		quirk_holder.remove_movespeed_modifier(/datum/movespeed_modifier/spacer)
+		quirk_holder.remove_status_effect(/datum/status_effect/spacer)
+
+		var/mob/living/carbon/human/human_quirker = quirk_holder
+		human_quirker.set_mob_height(HUMAN_HEIGHT_MEDIUM)
+		human_quirker.physiology.pressure_mod /= 0.8
+
+/datum/quirk/spacer_born/proc/start_tracking()
+	RegisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(moved_z))
+	check_z(quirk_holder, get_turf(quirk_holder), skip_timers = TRUE)
+	// Yes it's assumed, for planetary maps, that you start at gravity sickness.
+
+/// Proc to check quirk holder moves to determine when we should start tracking
+/datum/quirk/spacer_born/proc/arrived_at_station(mob/living/source, atom/old_loc)
+	SIGNAL_HANDLER
+
+	var/new_z = source.loc?.z
+	if(is_station_level(new_z) || is_mining_level(new_z))
+		UnregisterSignal(source, COMSIG_MOVABLE_MOVED)
+		start_tracking()
```

Comment:
why do you use Moved for this? it feels like you have this backward

Reply:
Figured moved would result in it starting tracking as soon as possible vs in gameplay, it doesn't necessary need to check for planet stuff every time the mob moved (so moved z does the job)